modules/auth-on-llama-stack.adoc

```diff
@@ -0,0 +1,388 @@
+:_module-type: PROCEDURE
+
+[id="auth-on-llama-stack_{context}"]
+= Configuring Llama Stack with OAuth Authentication
+
+You can configure Llama Stack to enable Role-Based Access Control (RBAC) for model access using OAuth authentication on {productname-short}. The following example shows how to configure Llama Stack so that a vLLM model can be accessed by all authenticated users, while an OpenAI model is restricted to specific users. This is example uses keycloak to issue and validate tokens.
+
+Before starting, you must already have KeyCloak set up with the following parameters:
+
+* Realm: `llamastack-demo`
+* Client: `llamastack` with direct access grants enabled
+* Role: `inference_max` grants access to restricted models and a protocol mapper that adds realm roles to the access token under the claim name `llamastack_roles`
+* Two test users:
+** `user1` as a basic user with no assigned roles
+** `user2` as an advanced user assigned the `inference_max` role
+* The client secret generated by Keycloak must be saved as you will need it for token requests.
+
+This document assumes the Keycloak server is available at `https://my-keycloak-server.com`
```

Comment:
Should this be a note?

Reply:
Would something like "This **example/scenario** assumes the Keycloak server is available at `https://my-keycloak-server.com`" be better?